internal/storage/database.go

```diff
@@ -127,27 +166,100 @@ func (s *StorageInPostgres) Save(value string) (string, error) {
 	return hashKey, nil
 }
 
+// FindByUserUID - поиск ссылок по пользовательскому UID.
+func (s *StorageInPostgres) FindByUserUID(userUID string) ([]ShortHashURL, error) {
+	var output []ShortHashURL
+	// SQL-запрос на поиск URLs
+	query := `
+		SELECT short, original FROM urls WHERE user_uid = $1
+	`
+	urls, err := s.connectionToDB.Query(context.Background(), query, userUID)
+
+	if err != nil {
+		log.Printf("Failed to find original URL: %v\n", err)
+		return output, err
+	}
+
+	defer urls.Close()
+
+	// Итерируем по строкам результата
+	for urls.Next() {
+		var shortURL, originalURL string
+
+		// Чтение данных в переменные
+		err = urls.Scan(&shortURL, &originalURL)
+		if err != nil {
+			log.Printf("failed to scan row: %s", err)
+			return output, err
+		}
+
+		// Добавление URL в массив
+		output = append(output, ShortHashURL{
+			ShortHash:   shortURL,
+			OriginalURL: originalURL,
+		})
+	}
+
+	if urls.Err() != nil {
+		log.Printf("error after iterating rows: %s", urls.Err())
+	}
+
+	return output, nil
+}
+
+// DeleteByUser - удалить ссылку по пользовательскому UUID
+func (s *StorageInPostgres) DeleteByUser(shortsHashURL []string, userUID string) error {
+
+	// В кеш
+	for _, v := range shortsHashURL { // записываем удаляемый батч в кеш, для запроса GET /{id}, чтобы не обращатся к БД
+		cache[v] = true
+	}
+
+	// Создаем объект Batch
+	batch := &pgx.Batch{}
+
+	for _, shortHashURL := range shortsHashURL { // short - короткая ссылка
+		batch.Queue("UPDATE urls SET deleted = true WHERE short = $1 and user_uid = $2", shortHashURL, userUID)
+	}
+
+	batchResults := s.poolConnectionToDB.SendBatch(context.Background(), batch)
+	defer batchResults.Close() // Закрываем BatchResults после использования
+
+	// Обработка каждой команды в батче
+	for {
+		_, err := batchResults.Exec()
+		if err != nil {
+			log.Printf("Error executing batch command: %v", err)
+			return err
+		}
+	}
```

Comment:
Меня смущает бесконечный цикл. Возможно стоит все-таки указать условие выхода?
```suggestion
	for i := 0; i < len(shortsHashURL); i++ {
		_, err := batchResults.Exec()
		if err != nil {
			log.Printf("Error executing batch command: %v", err)
			return err
		}
	}
```

Reply:
Готово, поправил